rose/web/game.js

```diff
@@ -332,6 +337,30 @@ var ROSE = (function() {
         }
     }
 
+    function FinishLine(loader) {
+        this.texture = null;
+        this.timeleft = 0;
+        var self = this;
+        loader.load("res/end/final_flag.png", function(img) {
+            self.texture = img;
+        });
+    }
+
+    FinishLine.prototype.update = function(state) {
+        this.timeleft = state.timeleft;
```

Comment:
We should protect from negative value here, since negative value will draw the finish line incorrectly.

We can use Math.max(state.timeleft, 0)

Reply:
done